test/ci-operator-configresolver-integration/run.sh

```diff
@@ -9,16 +9,43 @@ set -o pipefail
 
 ROOTDIR=$(pwd)
 WORKDIR="$( mktemp -d )"
-trap "rm -rf ${WORKDIR}" EXIT
+trap 'rm -rf ${WORKDIR}' EXIT
 
-cd $WORKDIR
+cd "$WORKDIR"
 # copy registry to tmpdir to allow tester to modify registry
-cp -a $ROOTDIR/test/ci-operator-configresolver-integration/ tests
-ci-operator-configresolver -config tests/configs -log-level debug &
+cp -a "$ROOTDIR"/test/ci-operator-configresolver-integration/ tests
+ci-operator-configresolver -config tests/configs -log-level debug -cycle 2m &
 PID=$!
-if ! timeout 10s bash -c 2>/dev/null -- "until diff tests/expected/openshift-installer-release-4.2.json <(curl 'http://127.0.0.1:8080/config?org=openshift&repo=installer&branch=release-4.2'); do sleep 1; done" &>/dev/null; then
-    echo "diff: $(diff -ru tests/expected/openshift-installer-release-4.2.json <(curl 'http://127.0.0.1:8080/config?org=openshift&repo=installer&branch=release-4.2'))"
+# generation should be >= 1. 0 or an error means the config function has not run yet
+# shellcheck disable=SC2016
+if ! timeout 10s bash -c 2>/dev/null -- 'until [[ $(curl http://127.0.0.1:8080/generation 2>/dev/null) -ne 0 ]]; do sleep 1; done'; then
```

Comment:
nit: unless we care about the time taken to actually run the `curl` we could be a little simpler:

```sh
for (( i = 0; i < 10; i++ )); do
    if [[ "$(curl http://127.0.0.1:8080/generation 2>/dev/null)" -ne 0 ]]; then
        break
    fi
    if [[ "${i}" -eq 10 ]]; then
        echo "[ERROR] Timed out waiting for ci-operator-configresolver to load configuration for the first time"
       exit 1
    fi
done
```